extendr-api/src/wrapper/list.rs

```diff
@@ -405,6 +405,13 @@ impl Deref for List {
     }
 }
 
+// A borrowed list can be converted to an owned Robj by cloning the Robj
```

Comment:
We could define this globally for everything that supports the `GetSexp` trait.

```
impl<T : GetSexp> Into<Robj> for &T {
    fn into(self) -> Robj {
        self.get_robj().to_owned()
    }
}
```

This would cover all wrapper types.

Reply:
Unfortunately that gives me:

```
error[E0210]: type parameter `T` must be covered by another type when it appears before the first local type (`robj::Robj`)
   --> extendr-api/src/robj/into_robj.rs:582:6
    |
582 | impl<T :GetSexp> Into<Robj> for &T
    |      ^ type parameter `T` must be covered by another type when it appears before the first local type (`robj::Robj`)
    |
    = note: implementing a foreign trait is only possible if at least one of the types for which it is implemented is local, and no uncovered type parameters appear before that first local type
    = note: in this case, 'before' refers to the following order: `impl<..> ForeignTrait<T1, ..., Tn> for T0`, where `T0` is the first and `Tn` is the last
```

I guess we can't actually blanket implement for `T` when we only have a trait bound for `T` (which admittedly is a local trait, so I feel like Rust should allow this)

Reply:
Oh, it *also* gives:
```
error[E0119]: conflicting implementations of trait `std::convert::Into<robj::Robj>` for type `&_`
   --> extendr-api/src/robj/into_robj.rs:582:1
    |
582 | impl<T :GetSexp> Into<Robj> for &T
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: conflicting implementation in crate `core`:
            - impl<T, U> Into<U> for T
              where U: From<T>;
```

Reply:
Rust doesn't yet have specialisation selection of overlapping traits and may never have!

Next best would be to include it in `make_conversions`.

Reply:
Done.